common/modules/farm/farm_base_impl/src/base_traits_impl.rs

```diff
@@ -196,12 +196,12 @@ pub trait FarmContract {
     ) {
         let farm_token_mapper = sc.farm_token();
         for farm_position in farm_positions {
+            farm_token_mapper.require_same_token(&farm_position.token_identifier);
+
             if sc.is_old_farm_position(farm_position.token_nonce) {
                 continue;
             }
 
-            farm_token_mapper.require_same_token(&farm_position.token_identifier);
-
             let token_attributes: FarmTokenAttributes<<Self::FarmSc as ContractBase>::Api> =
                 farm_token_mapper.get_token_attributes(farm_position.token_nonce);
 
```

Comment:
Can we extract the increase/decrease of the user_farm_position outside of the for, to improve the gas consumption?

Reply:
I'll add it in the new PR.

Reply:
Actually, it can be used only for increase position, as decrease position function needs the entire looped payment to check the attributes and decrease for the original owner. I'll see if there is any improvement if we send the original_owner as a parameter, instead of the payment.